src/vm/gdbjit.cpp

```diff
@@ -287,17 +295,67 @@ void NotifyGdb::MethodCompiled(MethodDesc* MethodDescPtr)
     StackScratchBuffer scratch;
     const char* szModName = modName.GetUTF8(scratch);
     const char *szModulePath, *szModuleFile;
-    
     SplitPathname(szModName, szModulePath, szModuleFile);
-    
+
+
+    int length = MultiByteToWideChar(CP_UTF8, 0, szModuleFile, -1, NULL, 0);
+    if (length == 0)
+        return;
+    LPWSTR wszModuleFile = new (nothrow) WCHAR[length+1];
+    length = MultiByteToWideChar(CP_UTF8, 0, szModuleFile, -1, wszModuleFile, length);
+
+    if (length == 0)
+        return;
+
+    DWORD cCharsNeeded;
+    cCharsNeeded = GetEnvironmentVariableW(W("CORECLR_GDBJIT"), NULL, 0);
+    BOOL isUserDebug = FALSE;
+
+    if ((cCharsNeeded != 0) && (cCharsNeeded < MAX_LONGPATH))
+    {
+        LPWSTR wszModuleNames = new WCHAR[cCharsNeeded+1];
```

Comment:
You can use NewArrayHolder<WCHAR> wszModuleNames here and don't need to worry about delete'ing it at the end of the block.